hoot-core-test/src/test/cpp/hoot/core/elements/WayTest.cpp

```diff
@@ -58,6 +59,76 @@ class WayTest : public HootTestFixture
     w.removeNode(2);
     HOOT_STR_EQUALS("[3]{1, 3, 5}", w.getNodeIds());
   }
+
+  void runReplaceTest()
+  {
+    Way w(Status::Unknown1, -1, 10);
+
+    w.clear();
+    w.addNode(1);
+    w.addNode(2);
+    w.addNode(5);
+    w.replaceNode(2, 4);
+    HOOT_STR_EQUALS("[3]{1, 4, 5}", w.getNodeIds());
+
+    // closed ways
+    w.clear();
+    w.addNode(1);
+    w.addNode(2);
+    w.addNode(1);
+    w.replaceNode(2, 4);
+    HOOT_STR_EQUALS("[3]{1, 4, 1}", w.getNodeIds());
+
+    // replacement node already exists - Note that Way::replaceNode does allow duplicated way nodes
```

Comment:
These newly added tests illustrate some of the problems that can occur when `Way::replaceNode` is called in certain ways. Ultimately decided not to change the behavior of that method.